### PR TITLE
Cluster the cluster data

### PR DESCRIPTION
**Pull request by ialarmedalien**
Part 1 of the changes in [this old PR in the relation_engine_spec repo](https://github.com/kbase/relation_engine_spec/pull/144).

Merge all `cluster` fields in the `djornl_node` collection into a single field.
Update parser and tests accordingly.

- [x] I updated the README.md docs to reflect this change. -- N/A
- [x] This is not a breaking API change

### CODE REVIEW

File: importers/djornl/parser.py [hunk 1]
```diff
@@ -43,15 +46,15 @@ def _configure(self):
 
         _CLUSTER_BASE = os.path.join(configuration['ROOT_DATA_PATH'], 'cluster_data')
         configuration['_CLUSTER_PATHS'] = {
-            'cluster_I2': os.path.join(
+            'markov_i2': os.path.join(
```
Comment (ialarmedalien) [inline importers/djornl/parser.py:49]: rename these to something more informative

File: importers/djornl/parser.py [hunk 2]
```diff
@@ -202,3 +220,32 @@ def load_data(self):
...
         self.save_dataset(self.load_cluster_data())
+
+    def check_data_delta(self):
+        edge_data = self.load_edges()
+        node_metadata = self.load_node_metadata()
+        clusters = self.load_cluster_data()
+
+        self.check_deltas(edge_data=edge_data, node_metadata=node_metadata, cluster_data=clusters)
+
+    def check_deltas(self, edge_data={}, node_metadata={}, cluster_data={}):
```
Comment (ialarmedalien) [inline importers/djornl/parser.py:231]: brief dataset summary for sanity checking

File: importers/test/test_djornl_parser.py [hunk 3]
```diff
@@ -100,33 +114,38 @@ def test_load_valid_edge_data(self):
...
         edge_data = parser.load_edges()
-        self.assertEqual(
-            edge_data,
-            self.json_data["load_edges"]
-        )
+        expected = self.json_data["load_edges"]
+
+        for data_structure in [edge_data, expected]:
+            for k in data_structure.keys():
+                data_structure[k] = sorted(data_structure[k], key=lambda n: n['_key'])
```
Comment (ialarmedalien) [inline importers/test/test_djornl_parser.py:122]: order data as it won't necessarily be sorted when coming out of the parser

File: spec/collections/djornl/djornl_node.yaml [hunk 4]
```diff
@@ -13,21 +17,15 @@ schema:
...
-      examples: [27]
+    clusters:
+      type: array
+      title: Clusters
+      description: Clusters to which the node has been assigned
+      items:
+        type: string
+        format: regex
+        pattern: ^\w+:\d+$
+      examples: [["markov_i2:1", "markov_i4:5"], ["markov_i6:3"]]
```
Comment (ialarmedalien) [inline spec/collections/djornl/djornl_node.yaml:28]: The important bit

File: spec/test/stored_queries/test_djornl.py [hunk 5]
```diff
@@ -81,108 +84,98 @@ def check_expected_results(self, description, response, expected):
...
+    #       }
+    #     }
+    #   }
+    # }
+    # if param_value is an array, join the array entities with "__"
     # results are in the form {"nodes": [...], "edges": [...]}
     # nodes are represented as a list of node[_key]
     # edges are objects with keys _to, _from, edge_type and score
 
-    def test_fetch_phenotypes_no_results(self):
```
Comment (ialarmedalien) [inline spec/test/stored_queries/test_djornl.py:97]: the queries with no results have been merged in with the other tests

File: spec/stored_queries/djornl/djornl_fetch_clusters.yaml [hunk 6]
```diff
@@ -2,25 +2,13 @@ name: djornl_fetch_clusters
...
-      title: Cluster I6 IDs
-      description: Cluster I6 IDs to locate
-      items: {type: integer}
-      examples: [[666], [999, 333]]
-      default: []
+    cluster_ids:
+      title: Cluster IDs
+      description: Cluster IDs, in the form "clustering_system_name:cluster_id"
+      items: {type: string}
+      examples: [['markov_i2:5', 'markov_i6:2'],['markov_i6:1']]
```
Comment (jayrbolton) [inline spec/stored_queries/djornl/djornl_fetch_clusters.yaml:11]: Should this be an object so we don't have to parse these entries?

Comment (jayrbolton) [reply spec/stored_queries/djornl/djornl_fetch_clusters.yaml:11]: I guess if the client is using string parameters like "markov_i2:5" then it doesn't matter

**Pull request merged**